build/templates/session.py.mako

```diff
@@ -191,7 +198,7 @@ constructor_params = helper.filter_parameters(init_function, helper.ParameterUsa
 % if attributes:
 
 % endif
-    def __init__(self, repeated_capability_list, ${config['session_handle_parameter_name']}, library, encoding, freeze_it=False):
+    def __init__(self, repeated_capability_list, ${config['session_handle_parameter_name']}, library, encoding, freeze_it=False, all_channels_in_session=None):
```

Comment:
From a logic perspective, it makes more sense for the new parameter `all_channels_in_session` to be adjacent to `repeated_capability_list` than to append it at the end.

Reply:
Appending it at the end has the benefit of preserving backward compatibility as well as the ability to give the parameter a default value (i.e. None). However, I figured that backward compatibility might not matter much here since `_SessionBase` is a private class, so I've moved the new parameter to after `repeated_capability_list` in my latest commit.